rfcs/select-component.md

```diff
@@ -0,0 +1,191 @@
+# Select Component
+
+### Exports
+
+* `StatefulSelect`
+* `StatefulSelectContainer`
+* `Select`
+* `StyledSelectInput`
+* `StyledOption`
+* `StyledDropDown`
+* `StyledLabel`
+* `StyledHint`
+
+### `Select` and `StatefulSelect` API
+
+* `type: TYPE.SEARCH | TYPE.SELECT`:
+  type of component to be in select or search mode (with lookup input)
+* `selectedOptions: Array<Object>`:
+  Current selected options. Every option object has `id: string` and `label: React$Node`. Label is defaulted to display for selected option, otherwise see `getSelectedOptionLabel` method
```

Comment:
I think a better type for this might be `string | Array<string>`
- For a single select, forcing people to use an array seems like an annoyance
- The selectedOptions will usually be passed in from the state of the parent form, and that state usually only includes keys, not labels. People will need to map the selected keys back to the corresponding option object, which is extra work for them – we should automatically do that internally. `string | Array<string>` also seems closer to what people expect from a `value` prop on a input.

Reply:
Native `Select` has imperfect interface. It's controversial. For single option it contain it in `value`, but for multiple options it is in `selectedOptions` and in totally different format. So I wouldn't reference to native Select as good practice. So I unified it to be always Array, so people no need to guess, what is the format of output. 
Also selectedOptions cannot come for Select from outside (parent) as they are a value of it's internal state (like for any input) and lots of routines based on them, if they always come from outside we cannot be sure they still support the same format, and more markup will be involved in re-rendering.
Mapping back is a concern, but if only `label` doesn't contain what is needed, and usually it does contain. You can look at `onSelect\onUnSelect` functions, which send back both `id` and `label`, so mapping is not needed unless we have data in other arrays or fields. `label` itself is a `React$Node`, which maybe anything valid: string, component, tag, etc